tests/codegen/riscv-abi/riscv64-lp64d-abi.rs

```diff
@@ -1,10 +1,16 @@
-//
-//@ compile-flags: -C no-prepopulate-passes
-//@ only-riscv64
-//@ only-linux
+//@ compile-flags: -C no-prepopulate-passes --target riscv64gc-unknown-linux-gnu
+//@ needs-llvm-components: riscv64
```

Comment:
Some minor tweaks:
```suggestion
//@ compile-flags: -O -C no-prepopulate-passes --target riscv64gc-unknown-linux-gnu
//@ needs-llvm-components: riscv
```
Optimizations can affect which attributes are added (`noundef` is one of them). But optimizations are enabled in these tests by default, so the test runs fine...until it runs on the no-opt builder and fails because the output is different. Specifying `-O` will make sure optimizations are consistently enabled. (I [always forget](https://github.com/rust-lang/rust/pull/122050/commits/f18c2f83e99ace7b9ad94a8204ff5fd68807ef34) to do this until bors tells me.)

The LLVM component name is `riscv` rather than `riscv64`. There's not really any way to tell what the component name is supposed to be; I confirmed it by looking at the output of `llvm-config --components` in my local LLVM. We should probably have a tidy check for this.

Reply:
Opened #125472 to have tidy lint for this, it's a bit of a footgun that `riscv` works and `riscv64` just ignores the test